src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,85 @@
-import React from 'react';
+/* eslint-disable react/jsx-one-expression-per-line */
+import React, { useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
 export const TodoModal: React.FC = () => {
+  const [isLoading, setLoading] = useState(false);
+  const selectedTodo = useAppSelector(state => state.currentTodo);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const dispatch = useAppDispatch();
+
+  async function getUsersFromServer() {
+    setLoading(true);
+
+    try {
+      const user = await getUser(selectedTodo?.userId || null);
+
+      setSelectedUser(user);
+    } catch (error) {
+      window.console.log('Error: ', error);
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  const closeModal = () => {
+    dispatch(currentTodoActions.removeTodo());
+  };
+
+  useEffect(() => {
```

Comment:
The 'useEffect' hook is missing dependencies. It currently runs only once like 'componentDidMount', but if 'getUsersFromServer' is expected to run again when certain props or state change, you should include those variables in the dependency array.